smartsim/builders/ensemble.py

```diff
@@ -31,22 +31,38 @@
 import itertools
 import os
 import os.path
+import random
 import typing as t
+from dataclasses import dataclass
 
+from smartsim._core.generation.operations.ensemble_operations import (
+    EnsembleConfigureOperation,
+    EnsembleFileSysOperationSet,
+)
 from smartsim.builders.utils import strategies
 from smartsim.builders.utils.strategies import ParamSet
 from smartsim.entity import entity
 from smartsim.entity.application import Application
-from smartsim.entity.files import EntityFiles
 from smartsim.launchable.job import Job
-from smartsim.settings.launch_settings import LaunchSettings
 
 if t.TYPE_CHECKING:
     from smartsim.settings.launch_settings import LaunchSettings
 
 
+@dataclass(frozen=True)
+class FileSet:
+    """
+    Represents a relationship between a parameterized set of arguments and the configuration file.
```

Comment:
```suggestion
    Specifies a set of parameterized arguments and the related configuration file.
```

feels awkward to read "represents"... i wonder if we can say what it does more directly